executor/postgreshandler.py

```diff
@@ -0,0 +1,468 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import datetime
+import hashlib
+from typing import Generator, List, Optional, Tuple
+
+import numpy as np
+import psycopg2
+import psycopg2.extras
+from jina import Document, DocumentArray
+from jina.logging.logger import JinaLogger
+from psycopg2 import pool  # noqa: F401
+
+
+def doc_without_embedding(d: Document):
+    new_doc = Document(d, copy=True)
+    new_doc.ClearField('embedding')
+    return new_doc.SerializeToString()
+
+
+SCHEMA_VERSION = 2
+SCHEMA_VERSIONS_TABLE_NAME = 'schema_versions'
+
+
+class PostgreSQLHandler:
+    """
+    Postgres Handler to connect to the database and
+     can apply add, update, delete and query.
+
+    :param hostname: hostname of the machine
+    :param port: the port
+    :param username: the username to authenticate
+    :param password: the password to authenticate
+    :param database: the database name
+    :param collection: the collection name
+    :param dry_run: If True, no database connection will be build
+    :param partitions: the number of shards to
+    distribute the data (used when rolling update on Searcher side)
+    :param args: other arguments
+    :param kwargs: other keyword arguments
+    """
+
+    def __init__(
+            self,
+            hostname: str = '127.0.0.1',
+            port: int = 5432,
+            username: str = 'default_name',
+            password: str = 'default_pwd',
+            database: str = 'postgres',
+            table: Optional[str] = 'default_table',
+            max_connections: int = 5,
+            dump_dtype: type = np.float64,
+            dry_run: bool = False,
+            partitions: int = 128,
+            *args,
+            **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.logger = JinaLogger('psq_handler')
+        self.table = table
+        self.dump_dtype = dump_dtype
+        self.partitions = partitions
+        self.snapshot_table = 'snapshot'
```

Comment:
```suggestion
        self.snapshot_table = f'snapshot_{table}'
```

Reply:
To use different snapshot table name for each table

Reply:
We don't want snapshotting here, right? At least not in v1